File: Data Structures & Algorithms/two-integer-sum/submission-3.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        mydict = {}
        
        for i, n in enumerate(nums):
            complement = target - n
            if complement in mydict:
                return[mydict[complement], i]
            else:
                mydict[n] = i
        
            

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-3.cpp
class Solution {
public:
    int findMin(vector<int> &nums) {
        int l = 0;
        int r = nums.size() - 1;

        if (nums[l] < nums[r]){
            return nums[l];
        }

        int res = nums[0];

        while (l <= r) {
            int m = (r - l) / 2 + l;
            res = min(nums[m], res);

            if (nums[m] < nums[r]) {
                r = m - 1;
            }

            else {
                l = m + 1;
            }
        }
        return res;
    }
};